system_tests/resources/linux/scripts/test_net.py

```diff
@@ -44,13 +46,55 @@ def get_interface(mac):
     return interface
 
 
-def set_ip_address(interface, ip):
-    run(
-        'ip addr add {ip}/24 dev {interface}'.format(
-            ip=ip,
+def interface_has_ip(interface, ip):
+    with settings(
+        hide('warnings', 'running', 'stdout', 'stderr'),
+        warn_only=True,
+    ):
+        command = 'ip addr show {interface} | grep {ip}'.format(
             interface=interface,
+            ip=ip,
         )
-    )
+        result = run(command)
+    if result.return_code == 0:
+        return False
+    elif result.return_code == 1:
+        return True
+
+
+def set_ip_address(interface, ip):
+    for i in range(0, 30):
+        if i == 0:
```

Comment:
Having the `i ==0` case here seems weird. Why not just do this before the for loop & move the other logging steps after the logic?